File: app/src/main/java/com/fittrack/data/entity/WorkoutRoutine.kt
package com.fittrack.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "workout_routines",
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = ["id"],
            childColumns = ["userId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("userId")]
)
data class WorkoutRoutine(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val userId: Int,
    val name: String,
    val lastPerformedAt: Long? = null,
    val timesPerformed: Int = 0,
    val createdAt: Long = System.currentTimeMillis()
)

File: ios/P3Fitness/Models/WorkoutRoutine.swift
import SwiftData
import Foundation

@Model
final class WorkoutRoutine {
    var id: Int
    var userId: Int
    var name: String
    var lastPerformedAt: Date?
    var timesPerformed: Int
    var createdAt: Date

    @Relationship(deleteRule: .cascade) var routineExercises: [RoutineExercise] = []

    init(userId: Int, name: String) {
        self.id = Int(Date().timeIntervalSince1970 * 1000) % 1_000_000_000
        self.userId = userId
        self.name = name
        self.timesPerformed = 0
        self.createdAt = Date()
    }
}
